datafusion/physical-expr/src/aggregate/array_agg_ordered.rs

```diff
@@ -228,11 +231,13 @@ impl Accumulator for OrderSensitiveArrayAggAccumulator {
                 .iter()
                 .map(|sort_expr| sort_expr.options)
                 .collect::<Vec<_>>();
-            self.values = merge_ordered_arrays(
+            let (new_values, new_orderings) = merge_ordered_arrays(
```

Comment:
Since we no longer rely on outside mechanism for merging, During merging we need to merge orderings (additional to values).